### PR TITLE
Add recognize for google client options

### LINKED ISSUE #197: google_client_options is an unrecognized option
The new `:google_client_options` exposed via the constructor is not listed as a recognized option, which leads to a warning (despite it actually being recognized). Luckily it's still used, but it's a bit noisy.

### PR DESCRIPTION
**Pull request by selmanj**
Cleaned up the recognizes option a bit and added :google_client_options to suppress warning.

Fixes #197

### CODE REVIEW

**Comment by icco** [comment]
Could you sort all of the recognizes lists so they are similar?

**Comment by selmanj** [comment]
@icco Sure, done.

File: lib/fog/google/pubsub.rb [hunk 1]
```diff
@@ -5,8 +5,17 @@ class Pubsub < Fog::Service
...
-                 :app_name, :app_version, :google_json_key_location, :google_json_key_string
+      recognizes(
+        :google_client_email,
+        :google_key_location,
+        :google_key_string,
+        :google_client,
+        :google_client_options,
+        :app_name,
+        :app_version,
+        :google_json_key_location,
```
Comment (icco) [inline lib/fog/google/pubsub.rb]: This ones out of order

Comment (selmanj) [inline lib/fog/google/pubsub.rb]: oh geez, how did i miss that

**Pull request merged**